src/credential/issuance/05-authorize-access.ts

```diff
@@ -103,12 +103,15 @@ export const authorizeAccess: AuthorizeAccess = async (
     },
     body: authorizationRequestFormBody.toString(),
   })
-    .then(hasStatus(200))
+    .then(hasStatusOrThrow(200, IssuerResponseError))
     .then((res) => res.json())
     .then((body) => TokenResponse.safeParse(body));
 
   if (!tokenRes.success) {
-    throw new ValidationFailed(tokenRes.error.message);
+    throw new ValidationFailed({
+      message: "Token Response validation failed",
+      reason: tokenRes.error.message,
+    });
```

Comment:
I see it duplicated many times. What do you think about writing a method for this?